{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare unassembled MAC and MIC libraries by k-mers\n",
    "\n",
    "For k-mer comparison we want to downsample reads to have similar coverage, and to compare R1 vs R2 reads as a sanity check, as [recommended in Kat docs](https://kat.readthedocs.io/en/latest/walkthrough.html#comparing-r1-v-r2-in-an-illumina-pe-dataset).\n",
    "\n",
    "Reads per library: Lmag MAC and MIC, two different sorting runs sequenced as separate libraries per nucleus type, with additional resequencing per library.\n",
    "\n",
    "Target | FACS experiments | Library | Reads    | Reseq  | Reseq reads\n",
    "-------|------------------|---------|----------|--------|-------------\n",
    "MAC    | 273-5U, 273-6U   | 4519.A  | 45451157 | 4615.A | 95684586\n",
    "MIC    | 273-5L, 273-6L   | 4519.B  | 47847972 | 4615.B | 93393507\n",
    "MAC    | 279-2U           | 4519.C  | 34013629 | 4615.C | 93084490\n",
    "MIC    | 279-2L           | 4519.D  | 44700862 | 4615.D | 94529508\n",
    "\n",
    "Final coverage is about 100x for ~280M reads pooled per library. Downsample of 90M reads per sample would therefore represent about 30x coverage."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "java -ea -Xmx200m -cp /ebio/ag-swart/home/kbseah/anaconda3/envs/bbmap/opt/bbmap-38.22-0/current/ jgi.ReformatReads threads=8 in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_A_R1_ktrim_qtrim28.fq.gz in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_A_R2_ktrim_qtrim28.fq.gz samplereadstarget=150000000 out=reads_downsampled/4615_A_R1_ktrim_qtrim28.150M.fq.gz out2=reads_downsampled/4615_A_R2_ktrim_qtrim28.150M.fq.gz\n",
      "Executing jgi.ReformatReads [threads=8, in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_A_R1_ktrim_qtrim28.fq.gz, in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_A_R2_ktrim_qtrim28.fq.gz, samplereadstarget=150000000, out=reads_downsampled/4615_A_R1_ktrim_qtrim28.150M.fq.gz, out2=reads_downsampled/4615_A_R2_ktrim_qtrim28.150M.fq.gz]\n",
      "\n",
      "Set threads to 8\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t344084052 reads          \t41313938496 bases\n",
      "Output:                 \t300000000 reads (87.19%) \t36021034004 bases (87.19%)\n",
      "\n",
      "Time:                         \t1370.358 seconds.\n",
      "Reads Processed:        344m \t251.09k reads/sec\n",
      "Bases Processed:      41313m \t30.15m bases/sec\n",
      "java -ea -Xmx200m -cp /ebio/ag-swart/home/kbseah/anaconda3/envs/bbmap/opt/bbmap-38.22-0/current/ jgi.ReformatReads threads=8 in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_B_R1_ktrim_qtrim28.fq.gz in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_B_R2_ktrim_qtrim28.fq.gz samplereadstarget=150000000 out=reads_downsampled/4615_B_R1_ktrim_qtrim28.150M.fq.gz out2=reads_downsampled/4615_B_R2_ktrim_qtrim28.150M.fq.gz\n",
      "Executing jgi.ReformatReads [threads=8, in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_B_R1_ktrim_qtrim28.fq.gz, in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_B_R2_ktrim_qtrim28.fq.gz, samplereadstarget=150000000, out=reads_downsampled/4615_B_R1_ktrim_qtrim28.150M.fq.gz, out2=reads_downsampled/4615_B_R2_ktrim_qtrim28.150M.fq.gz]\n",
      "\n",
      "Set threads to 8\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t335053580 reads          \t38909116338 bases\n",
      "Output:                 \t300000000 reads (89.54%) \t34838089973 bases (89.54%)\n",
      "\n",
      "Time:                         \t1327.897 seconds.\n",
      "Reads Processed:        335m \t252.32k reads/sec\n",
      "Bases Processed:      38909m \t29.30m bases/sec\n",
      "java -ea -Xmx200m -cp /ebio/ag-swart/home/kbseah/anaconda3/envs/bbmap/opt/bbmap-38.22-0/current/ jgi.ReformatReads threads=8 in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_C_R1_ktrim_qtrim28.fq.gz in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_C_R2_ktrim_qtrim28.fq.gz samplereadstarget=150000000 out=reads_downsampled/4615_C_R1_ktrim_qtrim28.150M.fq.gz out2=reads_downsampled/4615_C_R2_ktrim_qtrim28.150M.fq.gz\n",
      "Executing jgi.ReformatReads [threads=8, in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_C_R1_ktrim_qtrim28.fq.gz, in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_C_R2_ktrim_qtrim28.fq.gz, samplereadstarget=150000000, out=reads_downsampled/4615_C_R1_ktrim_qtrim28.150M.fq.gz, out2=reads_downsampled/4615_C_R2_ktrim_qtrim28.150M.fq.gz]\n",
      "\n",
      "Set threads to 8\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t339923000 reads          \t39978432434 bases\n",
      "Output:                 \t300000000 reads (88.26%) \t35283276378 bases (88.26%)\n",
      "\n",
      "Time:                         \t1345.283 seconds.\n",
      "Reads Processed:        339m \t252.68k reads/sec\n",
      "Bases Processed:      39978m \t29.72m bases/sec\n",
      "java -ea -Xmx200m -cp /ebio/ag-swart/home/kbseah/anaconda3/envs/bbmap/opt/bbmap-38.22-0/current/ jgi.ReformatReads threads=8 in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_D_R1_ktrim_qtrim28.fq.gz in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_D_R2_ktrim_qtrim28.fq.gz samplereadstarget=150000000 out=reads_downsampled/4615_D_R1_ktrim_qtrim28.150M.fq.gz out2=reads_downsampled/4615_D_R2_ktrim_qtrim28.150M.fq.gz\n",
      "Executing jgi.ReformatReads [threads=8, in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_D_R1_ktrim_qtrim28.fq.gz, in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_D_R2_ktrim_qtrim28.fq.gz, samplereadstarget=150000000, out=reads_downsampled/4615_D_R1_ktrim_qtrim28.150M.fq.gz, out2=reads_downsampled/4615_D_R2_ktrim_qtrim28.150M.fq.gz]\n",
      "\n",
      "Set threads to 8\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t345096192 reads          \t40473251800 bases\n",
      "Output:                 \t300000000 reads (86.93%) \t35184250632 bases (86.93%)\n",
      "\n",
      "Time:                         \t1358.844 seconds.\n",
      "Reads Processed:        345m \t253.96k reads/sec\n",
      "Bases Processed:      40473m \t29.79m bases/sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source activate bbmap\n",
    "for i in A B C D; do\n",
    "    reformat.sh threads=8 \\\n",
    "      in=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_${i}_R1_ktrim_qtrim28.fq.gz \\\n",
    "      in2=/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_${i}_R2_ktrim_qtrim28.fq.gz \\\n",
    "      samplereadstarget=150000000 \\\n",
    "      out=reads_downsampled/4615_${i}_R1_ktrim_qtrim28.150M.fq.gz \\\n",
    "      out2=reads_downsampled/4615_${i}_R2_ktrim_qtrim28.150M.fq.gz\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# COMPARE R1 R2 OF SAME LIBRARY\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "\n",
    "for i in B C D; do\n",
    "    kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC_4615_${i}_150M_R1_v_R2 \\\n",
    "     reads_downsampled/4615_${i}_R1_ktrim_qtrim28.150M.fq.gz \\\n",
    "     reads_downsampled/4615_${i}_R2_ktrim_qtrim28.150M.fq.gz &> kat-comp_k-19_LmagMAC_4615_${i}_150M_R1_v_R2.log\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    ". | A  | B  | C  | D |\n",
    "--|----|----|----|---|\n",
    "A | -  | x  | o  | x |\n",
    "B |    | -  | x  | o | \n",
    "C |    |    | -  | x |\n",
    "D |    |    |    | - |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# COMPARE DIFFERENT LIBRARIES FROM MAC vs MAC\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC-LmagMAC_4615_A_v_C_150M \\\n",
    " 'reads_downsampled/4615_A_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    " 'reads_downsampled/4615_C_R?_ktrim_qtrim28.150M.fq.gz' &> kat-comp_k-19_LmagMAC-LmagMAC_4615_A_v_C_150M.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# COMPARE DIFFERENT LIBRARIES FROM MIC vs MIC\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC-LmagMAC_4615_B_v_D_150M \\\n",
    " 'reads_downsampled/4615_B_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    " 'reads_downsampled/4615_D_R?_ktrim_qtrim28.150M.fq.gz' &> kat-comp_k-19_LmagMIC-LmagMIC_4615_B_v_D_150M.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# CROSS COMPARE MIC vs MAC\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC-LmagMAC_4615_A_v_B_150M \\\n",
    " 'reads_downsampled/4615_A_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    " 'reads_downsampled/4615_B_R?_ktrim_qtrim28.150M.fq.gz' &> kat-comp_k-19_LmagMAC-LmagMIC_4615_A_v_B_150M.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# CROSS COMPARE MIC vs MAC\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC-LmagMAC_4615_C_v_D_150M \\\n",
    " 'reads_downsampled/4615_C_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    " 'reads_downsampled/4615_D_R?_ktrim_qtrim28.150M.fq.gz' &> kat-comp_k-19_LmagMAC-LmagMIC_4615_C_v_D_150M.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# CROSS COMPARE MIC vs MAC\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC-LmagMAC_4615_A_v_D_150M \\\n",
    " 'reads_downsampled/4615_A_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    " 'reads_downsampled/4615_D_R?_ktrim_qtrim28.150M.fq.gz' &> kat-comp_k-19_LmagMAC-LmagMIC_4615_A_v_D_150M.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# CROSS COMPARE MIC vs MAC\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "kat comp -t 16 -m 19 -n -o kat-comp_k-19_LmagMAC-LmagMAC_4615_B_v_C_150M \\\n",
    " 'reads_downsampled/4615_B_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    " 'reads_downsampled/4615_C_R?_ktrim_qtrim28.150M.fq.gz' &> kat-comp_k-19_LmagMAC-LmagMIC_4615_B_v_C_150M.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# ASSEMBLY SPECTRUM Illumina downsampled vs MAC-assem\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "REF=/ebio/abt2_projects/ag-swart-loxodes/assembly/flye-comb_LmagMAC/assembly.fasta\n",
    "\n",
    "kat comp -t 16 -m 19 \\\n",
    "  -o kat-comp_k-19_LmagMAC_4615_A_150M_v_flye-comb_LmagMAC \\\n",
    "  'reads_downsampled/4615_A_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    "  $REF \\\n",
    "  &> kat-comp_k-19_LmagMAC_4615_A_150M_v_flye-comb_LmagMAC.log\n",
    "\n",
    "kat comp -t 16 -m 19 \\\n",
    "  -o kat-comp_k-19_LmagMAC_4615_B_150M_v_flye-comb_LmagMAC \\\n",
    "  'reads_downsampled/4615_B_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    "  $REF \\\n",
    "  &> kat-comp_k-19_LmagMAC_4615_B_150M_v_flye-comb_LmagMAC.log\n",
    "\n",
    "kat comp -t 16 -m 19 \\\n",
    "  -o kat-comp_k-19_LmagMAC_4615_C_150M_v_flye-comb_LmagMAC \\\n",
    "  'reads_downsampled/4615_C_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    "  $REF \\\n",
    "  &> kat-comp_k-19_LmagMAC_4615_C_150M_v_flye-comb_LmagMAC.log\n",
    "\n",
    "kat comp -t 16 -m 19 \\\n",
    "  -o kat-comp_k-19_LmagMAC_4615_D_150M_v_flye-comb_LmagMAC \\\n",
    "  'reads_downsampled/4615_D_R?_ktrim_qtrim28.150M.fq.gz' \\\n",
    "  $REF \\\n",
    "  &> kat-comp_k-19_LmagMAC_4615_D_150M_v_flye-comb_LmagMAC.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "kat-comp_k-19_LmagMAC_4615_A_150M_R1_v_R2.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 7.03563e+07\n",
      " - Euclidean distance: 1.19481e+07\n",
      " - Cosine distance: 0.0302026\n",
      " - Canberra distance: 154.746\n",
      " - Jaccard distance: 0.256772\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 6.47774e+07\n",
      " - Euclidean distance: 1.04633e+07\n",
      " - Cosine distance: 0.0477699\n",
      " - Canberra distance: 155.121\n",
      " - Jaccard distance: 0.270385\n",
      "\n",
      "kat-comp_k-19_LmagMAC_4615_B_150M_R1_v_R2.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 7.81055e+07\n",
      " - Euclidean distance: 1.3431e+07\n",
      " - Cosine distance: 0.0436127\n",
      " - Canberra distance: 201.168\n",
      " - Jaccard distance: 0.290165\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 7.58525e+07\n",
      " - Euclidean distance: 1.2879e+07\n",
      " - Cosine distance: 0.0650533\n",
      " - Canberra distance: 201.756\n",
      " - Jaccard distance: 0.311907\n",
      "\n",
      "kat-comp_k-19_LmagMAC_4615_C_150M_R1_v_R2.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 6.52897e+07\n",
      " - Euclidean distance: 1.12342e+07\n",
      " - Cosine distance: 0.035104\n",
      " - Canberra distance: 161.503\n",
      " - Jaccard distance: 0.252582\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 6.12407e+07\n",
      " - Euclidean distance: 1.01093e+07\n",
      " - Cosine distance: 0.0428761\n",
      " - Canberra distance: 161.872\n",
      " - Jaccard distance: 0.259715\n",
      "\n",
      "kat-comp_k-19_LmagMAC_4615_D_150M_R1_v_R2.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 6.90508e+07\n",
      " - Euclidean distance: 1.23551e+07\n",
      " - Cosine distance: 0.0406512\n",
      " - Canberra distance: 172.136\n",
      " - Jaccard distance: 0.266591\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 6.41971e+07\n",
      " - Euclidean distance: 1.09381e+07\n",
      " - Cosine distance: 0.0471291\n",
      " - Canberra distance: 172.469\n",
      " - Jaccard distance: 0.271469\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for i in *R2.stats; do\n",
    "  echo $i\n",
    "  grep -A5 'Distance' $i\n",
    "  echo \"\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "kat-comp_k-19_LmagMAC_4615_A_150M_v_flye-comb_LmagMAC.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 3.58564e+08\n",
      " - Euclidean distance: 1.32904e+08\n",
      " - Cosine distance: 0.638038\n",
      " - Canberra distance: 978.458\n",
      " - Jaccard distance: 0.876324\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 3.63313e+08\n",
      " - Euclidean distance: 1.4367e+08\n",
      " - Cosine distance: 0.994682\n",
      " - Canberra distance: 980.492\n",
      " - Jaccard distance: 0.981311\n",
      "\n",
      "kat-comp_k-19_LmagMAC_4615_B_150M_v_flye-comb_LmagMAC.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 3.53088e+08\n",
      " - Euclidean distance: 1.36154e+08\n",
      " - Cosine distance: 0.703092\n",
      " - Canberra distance: 977.254\n",
      " - Jaccard distance: 0.88562\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 3.61002e+08\n",
      " - Euclidean distance: 1.43774e+08\n",
      " - Cosine distance: 0.993047\n",
      " - Canberra distance: 978.689\n",
      " - Jaccard distance: 0.978163\n",
      "\n",
      "kat-comp_k-19_LmagMAC_4615_C_150M_v_flye-comb_LmagMAC.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 3.47575e+08\n",
      " - Euclidean distance: 1.36286e+08\n",
      " - Cosine distance: 0.705045\n",
      " - Canberra distance: 977.973\n",
      " - Jaccard distance: 0.887359\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 3.61807e+08\n",
      " - Euclidean distance: 1.43681e+08\n",
      " - Cosine distance: 0.993522\n",
      " - Canberra distance: 979.722\n",
      " - Jaccard distance: 0.979271\n",
      "\n",
      "kat-comp_k-19_LmagMAC_4615_D_150M_v_flye-comb_LmagMAC.stats\n",
      "Distance between spectra 1 and 2 (all k-mers):\n",
      " - Manhattan distance: 3.45618e+08\n",
      " - Euclidean distance: 1.35808e+08\n",
      " - Cosine distance: 0.693012\n",
      " - Canberra distance: 977.049\n",
      " - Jaccard distance: 0.886744\n",
      "--\n",
      "Distance between spectra 1 and 2 (shared k-mers):\n",
      " - Manhattan distance: 3.60372e+08\n",
      " - Euclidean distance: 1.43782e+08\n",
      " - Cosine distance: 0.992481\n",
      " - Canberra distance: 978.356\n",
      " - Jaccard distance: 0.977302\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for i in *LmagMAC.stats; do\n",
    "  echo $i\n",
    "  grep -A5 'Distance' $i\n",
    "  echo \"\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# ASSEMBLY SPECTRUM Illumina complete vs MAC-assem\n",
    "set -e\n",
    "source activate /ebio/abt2_projects/ag-swart-loxodes/envs/kat\n",
    "REF=/ebio/abt2_projects/ag-swart-loxodes/assembly/flye-comb_LmagMAC/assembly.fasta\n",
    "\n",
    "kat comp -t 16 -m 19 \\\n",
    "  -o kat-comp_k-19_LmagMAC-comb_v_flye-comb_LmagMAC \\\n",
    "  '/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4519_A_R?_ktrim_qtrim28.fq.gz /ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4519_C_R?_ktrim_qtrim28.fq.gz /ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_A_R?_ktrim_qtrim28.fq.gz /ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_C_R?_ktrim_qtrim28.fq.gz' \\\n",
    "  $REF \\\n",
    "  &> kat-comp_k-19_LmagMAC-comb_v_flye-comb_LmagMAC.log\n",
    "\n",
    "\n",
    "kat comp -t 16 -m 19 \\\n",
    "  -o kat-comp_k-19_LmagMIC-comb_v_flye-comb_LmagMAC \\\n",
    "  '/ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4519_B_R?_ktrim_qtrim28.fq.gz /ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4519_D_R?_ktrim_qtrim28.fq.gz /ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_B_R?_ktrim_qtrim28.fq.gz /ebio/abt2_projects/ag-swart-loxodes/data/reads-trim/4615_D_R?_ktrim_qtrim28.fq.gz' \\\n",
    "  $REF \\\n",
    "  &> kat-comp_k-19_LmagMIC-comb_v_flye-comb_LmagMAC.log"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:bio-env]",
   "language": "python",
   "name": "conda-env-bio-env-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
